js/jsonld.js

```diff
@@ -1938,14 +1944,6 @@ jsonld.documentLoaders.xhr = function(options) {
   var queue = new jsonld.RequestQueue();
   var headers = buildHeaders(options.headers);
 
-  // use option or, by default, use Promise when its defined
-  var usePromise = ('usePromise' in options ?
-    options.usePromise : (typeof Promise !== 'undefined'));
-  if(usePromise) {
-    return queue.wrapLoader(function(url) {
-      return jsonld.promisify(loader, url);
-    });
-  }
```

Comment:
Needs a similar fix in the other loaders. I also suspect that the `wrapLoader` function needs to change to remove the `_usePromise` flag and treat every loader like it works with both a callback and a Promise (unless the internals of that queue could work with just a callback).

I don't think we can avoid mandating that anyone writing a loader must implement both callbacks and return Promises ... right? We can't automate how custom loaders will work, I don't think, can we? Isn't it too messy and unreliable?

I guess if we want to make it easier for devs to write custom loaders, we should add a helper function that will wrap a callback-only or Promise-only loader function appropriately to expose both mechanisms.

Thoughts?